File: apps/betty-server/src/trips/application/use-cases/get-trip-by-id/get-trip-by-id.use-case.ts
import { Inject, Injectable } from '@nestjs/common';
import { Trip } from '../../../domain/entities/trip.entity';
import {
  TripRepository,
  TripWithLocations,
} from '../../../domain/repositories/trip.repository';
import { TRIP_REPOSITORY } from '../../../infrastructure/ioc/symbols';

@Injectable()
export class GetTripByIdUseCase {
  constructor(
    @Inject(TRIP_REPOSITORY)
    private readonly tripRepository: TripRepository
  ) {}

  async execute(
    id: string,
    includeLocations = false
  ): Promise<Trip | TripWithLocations | null> {
    return this.tripRepository.findById(id, includeLocations);
  }
}

File: apps/betty_app/lib/_old/trips/application/use_cases/get_trip_by_id_use_case.dart
import '../../infrastructure/services/trip_api_service.dart';
import '../../domain/entities/trip.dart';

class GetTripByIdUseCase {
  final TripApiService _apiService;

  const GetTripByIdUseCase(this._apiService);

  Future<Trip?> execute(String id) async {
    return await _apiService.getTripById(id);
  }
}
